src/modules/wallet/api/updaters/HwAccountIndexUpdater.tsx

```diff
@@ -0,0 +1,52 @@
+import { useAtom } from 'jotai'
+import { useEffect, useMemo, useRef } from 'react'
+
+import { useWeb3React } from '@web3-react/core'
+
+import { useAppSelector } from 'legacy/state/hooks'
+
+import { getWeb3ReactConnection, HARDWARE_WALLETS } from '../../web3-react/connection'
+import { useWalletInfo } from '../hooks'
+import { hwAccountIndexAtom } from '../state'
+
+const indexChanged = true
+
+export function HwAccountIndexUpdater() {
+  const [hwAccountIndex, setHwAccountIndex] = useAtom(hwAccountIndexAtom)
+  const { chainId } = useWalletInfo()
+  const { connector, isActive } = useWeb3React()
+  const connectorRef = useRef(connector)
+  const selectedWallet = useAppSelector((state) => state.user.selectedWallet)
+
+  connectorRef.current = connector
+
+  const connectionType = useMemo(() => {
+    const connection = getWeb3ReactConnection(connector)
+
+    return connection.type
+  }, [connector])
+
+  /**
+   * Reactivate connector each time when account index is changed from HwAccountIndexSelector
+   * A hardware wallet connector should take into account the second parameter (indexChanged = true) for activate() method
+   */
+  useEffect(() => {
+    if (!isActive) return
+
+    const isHardWare = HARDWARE_WALLETS.includes(connectionType)
+
+    if (!isHardWare) return
+
+    console.debug('[Hardware wallet] account index changed', hwAccountIndex)
+    connectorRef.current?.activate(chainId, indexChanged)
+  }, [isActive, hwAccountIndex, connectionType, chainId])
+
+  useEffect(() => {
+    if (selectedWallet) return
+
+    console.debug('[Hardware wallet] reset account index to 0')
+    setHwAccountIndex(0)
+  }, [setHwAccountIndex, selectedWallet])
```

Comment:
Should we keep the debug statements?

Reply:
I would say yes. Because it's not repetitive action and at least for the beginning it will be helpful